akka-testkit-typed/src/main/scala/akka/testkit/typed/Effect.scala

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.testkit.typed
+
+import akka.actor.typed.{ ActorRef, Behavior, Props }
+import akka.annotation.{ ApiMayChange, DoNotInherit }
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+/**
+ * All tracked effects for the [[akka.testkit.typed.scaladsl.BehaviorTestKit]] and
+ * [[akka.testkit.typed.javadsl.BehaviorTestKit]] must extend this type.
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed abstract class Effect
+
+/**
+ * Factories/types for effects, each effect has a suitable equals and can be used to compare actual effects
+ * to expected ones.
+ */
+@ApiMayChange
+object Effect {
+
+  // FIXME is this one really a useful subgroup?
+  abstract class SpawnedEffect extends Effect
+
+  final case class Spawned[T](behavior: Behavior[T], childName: String, props: Props = Props.empty) extends SpawnedEffect
+
+  /**
+   * Java API:
```

Comment:
too complicated to separate this in scaladsl, javadsl?

Reply:
Hmm, I guess that is better. Maybe we could keep the actual classes/objects in the Scala API and have methods only on the Java side. (I did it like this because I thought it would then be easy to miss when adding factory methods, but I guess we won't really be adding new effects very often)